ui/src/components/SampleView/shapes.js

```diff
@@ -221,7 +221,7 @@ export function makePoint(x, y, id, color, type, name, strokeWidth) {
 export function makePoints(points, imageRatio) {
   const fabricPoints = [];
   for (const id in points) {
-    if (Object.hasOwn(points, id)) {
+    if ({}.hasOwnProperty.call(points, id)) {
```

Comment:
SIgnalled as bug in Pycharm, and caused malfunction until reverted.
@marcus-oscarsson May be a denepndency/version issue, or may require a different fix (though this one works in mock mode).

Reply:
Its not a bug, it has the same meaning (more or less) and that is also why it works :). But its always better to be consistent, we should in that case use [hasOwn](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/hasOwn)

Reply:
Hm, reading this a second time. I'm not sure that I understand. You write that hasOwnProperty works but not hasOwn and that you are reverting it (but to what point in time, when was it changed to hasOwn ?)

Reply:
Well, my editor (pycharm) signalled it as a bug (no such function), and changing it (unless there was another simultaneous change and I got confused) fixed a problem that gave me a white screen. Could it be a question of whi dependencies / version is installed?

Reply:
According to Pycharm git history, the commit by JB* 30/3/2022 "Merge branch 'mo-create-react-app' into jb-bootstrap5" has
    if ({}.hasOwnProperty.call(points, id)) {

whereas the commit by you 02/05/2022 "lint" introduced 
    if (Object.hasOwn(points, id)) {

I shall make a quick test and see what happens if I go back to 'hasOwn'

Reply:
I think I found it, it was a automatic linting run that changed this. The change is seemingly correct but there seems to be a subtle difference between how hasOwn and hasOwnProperty that perhaps breaks the code. I think you are right that using hasOwnProperty is the safest option until we have figured out what goes wrong (hasOwn seems to work here when I test)

Reply:
Well done :)

Reply:
So better to use **hasOwnProperty** then, again well spotted :)

Reply:
That would make sense.
I run FIrefox version 91.10.0 extended support release for OpenSUSE Leap 15.3, released on August 10, 2021, according to Wikipedia.